ecl/hql/hqlcollect.cpp

```diff
@@ -468,9 +480,69 @@ void FileSystemDirectory::populateChildren()
     }
 }
 
+void FileSystemDirectory::processDependencies(const char * rootPath)
+{
+    StringBuffer fullname;
+    StringBuffer searchPath(rootPath);
+    //Implement the node import rules.  Allow node_modules in this directory, or any parent directory
+    //and if present include the contents as dependencies.
+    for (;;)
+    {
+        addPathSepChar(fullname.clear().append(searchPath)).append(specialDependencyDirectory);
+        Owned<IFile> file = createIFile(fullname);
+        if (file->isDirectory() == fileBool::foundYes)
+        {
+            Owned<IDirectoryIterator> dir = file->directoryFiles(NULL, false, true);
+            processDependencies(dir);
+        }
+
+        //Now check if there is a node_modules directory in the parent directory
+        StringBuffer parentPath;
+        splitFilename(searchPath, &parentPath, &parentPath, nullptr, nullptr);
 
-void FileSystemDirectory::processDependencies(IPropertyTree * dependTree, const char * path)
+        removeTrailingPathSepChar(parentPath);
+        if (parentPath.isEmpty() || isRootDirectory(parentPath))
+            return;
+
+        searchPath.set(parentPath);
+    }
+}
+
+void FileSystemDirectory::processDependencies(IDirectoryIterator * dir)
 {
+    ForEach (*dir)
+    {
+        IFile &file = dir->query();
+        const char * filename = file.queryFilename();
+        const char * tail = pathTail(filename);
+        if (file.isDirectory() == fileBool::foundYes)
+        {
+            //If there is a directory within node_module, then treat that directory as a package
+            PackageDependency * depend = new PackageDependency(createIdAtom(tail), filename, false);
```

Comment:
Any possibilities with the package name atom clashing with atoms created elsewhere for other purposes?

Reply:
It wouldn't matter if they did.  It is an atom table of ids that can use pointer comparison for quick equality checking.